File: .NET/Demo_Common/Entity/ESLEntity.cs
using MessagePack;
using PageIndex = Demo_Common.Enum.PageIndex;
using Pattern = Demo_Common.Enum.Pattern;

namespace Demo_Common.Entity
{
    /// <summary>
    /// Tag entity
    /// </summary>
    [MessagePackObject]
    public class ESLEntity
    {
        [Key(0)]
        public string TagID { get; set; } = "";
        [Key(1)]
        public Pattern Pattern { get; set; } = Pattern.UpdateDisplay;
        [Key(2)]
        public PageIndex PageIndex { get; set; } = PageIndex.P0;
        [Key(3)]
        public bool R { get; set; } = false;
        [Key(4)]
        public bool G { get; set; } = false;
        [Key(5)]
        public bool B { get; set; } = false;
        [Key(6)]
        public int Times { get; set; } = 0;
        [Key(7)]
        public int Token { get; set; } = 0;
        [Key(8)]
        public string OldKey { get; set; } = "";
        [Key(9)]
        public string NewKey { get; set; } = "";
        [Key(10)]
        public string Base64String { get; set; } = "";
    }
}

File: Python/Entities/eslEntity.py
import msgpack
from dataclasses import dataclass, field
from Enums.pattern import Patterns
from Enums.pageIndex import PageIndexes

@dataclass
class ESLEntity:
    """
    ESL entity for eStation2
    """
    TagID: str = ""
    Pattern: Patterns = Patterns.UpdateDisplay
    PageIndex: PageIndexes = PageIndexes.P0
    R: bool = False
    G: bool = False
    B: bool = False
    Times: int = 0
    Token: int = 0
    CurrentKey: str = ""
    NewKey: str = ""
    Base64String: str = ""

    def to_msgpack(self) -> bytes:
        # Serializes the object to MessagePack format using key order
        data = [
            self.TagID,
            self.Pattern,
            self.PageIndex,
            self.R,
            self.G,
            self.B,
            self.Times,
            self.Token,
            self.CurrentKey,
            self.NewKey,
            self.Base64String
        ]
        return msgpack.packb(data)

    @staticmethod
    def from_msgpack(packed: bytes):
        # Deserializes MessagePack data to ESLEntity
        data = msgpack.unpackb(packed)
        return ESLEntity(
            TagID=data[0],
            Pattern=data[1],
            PageIndex=data[2],
            R=data[3],
            G=data[4],
            B=data[5],
            Times=data[6],
            Token=data[7],
            CurrentKey=data[8],
            NewKey=data[9],
            Base64String=[10]
        )